packages/core/test/input.test.ts

```diff
@@ -21,29 +31,55 @@ describe('input', () => {
       }
     });
 
+    interpret(machine, {
+      input: {
+        // @ts-expect-error
+        wrongCountProperty: 42
+      }
+    });
+
     interpret(machine, { input: { startCount: 42 } }).start();
 
     expect(spy).toHaveBeenCalledWith(42);
   });
 
   it('initial event should have input property', (done) => {
```

Comment:
this test case no longer tests what it should :wink:

Reply:
Yeah, we shouldn't rely on the event having the input data. That should be saved to `context` and read there as a best practice

Reply:
As a best practice - sure, that's probably for the better. As it stands though, this is also part of the init event (creating a nice symmetry with "done" event that contains `output` and we can't easily remove it from there). We should be careful when adjusting tests - we don't want to lose valuable tests and assertions. Even if we'd like to remove `input` from the init event then this whole test should be gone (or adjusted to document that there is only `type` property on the init event).

Reply:
Keeping `input` on the init event has also one pretty nice advantage - it will be easier to keep this information around in the inspector tools etc